python/graphstorm/gconstruct/utils.py

```diff
@@ -329,8 +547,8 @@ def _merge_arrs(arrs, tensor_path):
         for arr in arrs:
             out_arr[row_idx:(row_idx + arr.shape[0])] = arr[:]
             row_idx += arr.shape[0]
-        return out_arr
-    elif isinstance(arrs[0], HDF5Array):
+        return ExtNumpyWrapper(tensor_path, out_arr.shape, out_arr.dtype)
```

Comment:
Why not pass out_arr to ExtNumpyWrapper?

Reply:
we want to release the original numpy array and open a new one when it is used.